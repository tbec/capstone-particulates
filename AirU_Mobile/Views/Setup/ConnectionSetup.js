import React, {Component} from 'react';
import {Text, View, Image, TextInput, Button} from 'react-native';
import NavBar from '../../Components/NavBar'
import styles from '../../StyleSheets/Styles'

export default class ConnectionSetup extends Component<Props> {
    constructor(props) {
        super(props)
        // timer for faking
        this.updateTimer = this.updateTimer.bind(this);
        this.connectToWiFi = this.connectToWiFi.bind(this);
        let _timer = setInterval(this.updateTimer, 5000);

        this.state={bleConnected: false, MAC: null, wifiConnected: false, 
                    WiFiName: "", WiFiPassword: "", WiFiError: false, timer: _timer};
    }

    // used to fake connecting to BT
    updateTimer() {
        this.setState({MAC: 'MAC ADDRESS'});
      }

    connectToBluetooth() {
        // dummy code, make me actually work
    }

    connectToWiFi() {
        // if valid, navigate to Privacy. Otherwise mark as error
        if (this.state.WiFiName == "Utah" && this.state.WiFiPassword == "password") {
           this.props.navigation.navigate("Privacy");
        }
        else {
            this.setState({WiFiError: true})
        }
    }

    componentWillUnmount() {
        clearInterval(this.state.timer);
    }

    render() {
        var error, macAddress

        // after trying to connect will set state, renders error text if true
        if (this.state.WiFiError == true) {
            error = <Text style={{color: 'red', fontSize: 12}}>Could not connect to WiFi Network</Text>
        }
        else {
            error = <Text style={{color: 'red', fontSize: 12}}></Text>
        }

        // faking loading
        if (this.state.MAC != null) {
            macAddress=<Text>{this.state.MAC}</Text>
        }
        else {
            macAddress=<Image source={require('../../Resources/Loading.gif')} 
                                style={{width: 100, height: 100, alignContent: 'center', justifyContent: 'center'}}/>
        }
        
        return (
            <View style={styles.mainView}>
                <View style={{flex: 2}}>
                    <Text>Enable your Bluetooth and connect to the sensor. Once it is connected the device name will show below. 
                        Select your WiFi network to connect to, enter the password, then select 'Connect'.
                    </Text>
                </View>
                {/* BLE name goes here */}
                <View style={{flex: 2, jusifyContent: 'flex-start', alignContent: 'flex-start'}}>
                    {macAddress}
                </View>
                <View style={{flex: 2}}>
                    <Text>Confirm the MAC ID listed above matches the one on your sensor. If it is correct, 
                        enter WiFi information below to connect the sensor to your network.
                    </Text>
                </View>
                <View style={{flex: 3, alignContent: 'flex-start', justifyContent: 'flex-start',
                        paddingLeft: 30}}>
                    {/* network */}
                    <TextInput editable={true} keyboardType='default' 
                                autoCorrect={false} placeholder='SSID' secureTextEntry={false}
                                style={{borderWidth: 2, borderColor: 'black', 
                                width: '50%', height: 40, alignContent: 'center', justifyContent: 'center', paddingBottom: 10}}
                                onChangeText={(value) => {this.setState({WiFiName: value})}}
                                />
                    <Text/>
                    {/* password */}
                    <TextInput editable={true} keyboardType='default' 
                                autoCorrect={false} placeholder='Password' secureTextEntry={true} 
                                style={{borderWidth: 2, borderColor: 'black', 
                                width: '50%', height: 40, alignContent: 'center', justifyContent: 'center', paddingBottom: 10}}
                                onChangeText={(value) => {this.setState({WiFiPassword: value})}}
                                />
                </View>
                <View>
                    {/* connect button */}
                    <Button title="Connect"
                        onPress={() => this.connectToWiFi()}
                        disabled={(this.state.WiFiPassword == "" || this.state.WiFiName == "")}
                    />
                </View>
                <View style={{flex: 2}}>
                    {/* <Text>{this.state.WiFiName}</Text>
                    <Text>{this.state.WiFiPassword}</Text> */}
                    {error}
                </View>
                <NavBar navigation={this.props.navigation} previous='MountingSensor'/>
            </View>
        );
    }
}