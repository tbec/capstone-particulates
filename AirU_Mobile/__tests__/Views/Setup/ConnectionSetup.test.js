import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import ConnectionSetup from '../../../Views/Setup/ConnectionSetup'

// used to find testIDs in code
let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

test('Snapshot ConnectionSetup Setup', () => {
    const snap = renderer.create(<ConnectionSetup/>).toJSON();
    expect(snap).toMatchSnapshot();
})