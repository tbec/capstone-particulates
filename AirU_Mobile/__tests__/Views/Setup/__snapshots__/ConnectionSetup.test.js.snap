// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ConnectionSetup Setup 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 5,
            "flexDirection": "column",
            "padding": 40,
          },
          Object {
            "flex": 2,
            "flexDirection": "column",
            "paddingTop": 40,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../Resources/Setup_Connecting.png",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "140%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text>
        Enable your Bluetooth and connect to the sensor. Once it is connected the device name will show below. Select your WiFi network to connect to, enter the password, then select 'Connect'.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignContent": "flex-start",
          "flex": 2,
          "jusifyContent": "flex-start",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../Resources/Loading.gif",
          }
        }
        style={
          Object {
            "alignContent": "center",
            "height": 100,
            "justifyContent": "center",
            "width": 100,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text>
        Confirm the MAC ID listed above matches the one on your sensor. If it is correct, enter WiFi information below to connect the sensor to your network.
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignContent": "flex-start",
          "flex": 3,
          "justifyContent": "flex-start",
          "paddingLeft": 30,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="SSID"
        secureTextEntry={false}
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "white",
            "borderColor": "black",
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "paddingBottom": 10,
            "width": "50%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "white",
            "borderColor": "black",
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "paddingBottom": 10,
            "width": "50%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View>
      <View
        accessibilityRole="button"
        accessibilityStates={
          Array [
            "disabled",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "red",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Connect
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "red",
            "fontSize": 12,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignContent": "space-between",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "flex-end",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "transparent",
            "borderColor": "black",
            "flex": 1,
            "justifyContent": "flex-start",
            "padding": 10,
          }
        }
        testID="prevButton"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 40,
              },
              null,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï„’
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
